client/scala/src/main/scala/ai/verta/repository/Commit.scala

```diff
@@ -338,4 +338,78 @@ class Commit(
     else
       createCommit(message = message, diffs = diff.blobDiffs, commitBase = id)
   }
+
+  /** Generates a stream of outputs in this commit by walking through its folder tree.
+   *  The stream ends at the first failure, or when there are no folders left. If the commit is not saved, its only element is that Failure
+   *  @param walker the FolderWalker to process the folders in the folder tree.
+   *  @return a stream of Try's of WalkOutput.
+   */
+  def walk[T](walker: FolderWalker[T])(implicit ec: ExecutionContext): Stream[Try[T]] = {
+    if (!saved)
+      Stream(Failure(new IllegalCommitSavedStateException("Commit must be saved before it can be walked")))
+    else
+      continueWalk(getFolder(PathList(Nil)).get, walker)
+  }
+
+  /** Get the folder corresponding to a path in list form
+   *  @param location location
+   *  @return the folder, if succeeds
+   */
+  def getFolder(location: PathList)(implicit ec: ExecutionContext): Try[Folder] = {
+    clientSet.versioningService.GetCommitComponent2(
+      commit_sha = id.get,
+      repository_id_repo_id = repo.id,
+      location = if (location.components.length > 0) Some(location.components) else None
+    ).flatMap(r => {
+        val folderPath = location.path
+        val responseFolder = r.folder
+
+        val folderNames = responseFolder
+        .flatMap(_.sub_folders) // Option[List[VersioningFolderElement]]
```

Comment:
Fix indentation all cross this PR. Not sure if this was automatic, but doesn't look like the standard indentation.

Reply:
Huh weird. I'll manually align the method chaining. See if it looks better.

Reply:
What formatter do you use?

Reply:
None actually. I was just using the default scala plugin of Atom, along with an auto-indent plugin, which always seems to fail me, so I have to manually fix the indentation a lot of times. Is there one that you would recommend for Atom?